### PR TITLE
fix(docker): use docker image with current sources

### LINKED ISSUE #602: Build docker image according to current source tree
## Description of the problem, including code/CLI snippet
Currently in the docker image, the package just gets installed from PyPI, which isn't the usual way to provide docker files in a source repository.

## Expected Behavior
Dockerfile get's build into wheel, which get's installed into multi-layer build

## Actual Behavior
Package installed from PyPI

### PR DESCRIPTION
**Pull request by max-wittig**
Fixes #602

### CODE REVIEW

File: Dockerfile [hunk 1]
```diff
@@ -0,0 +1,15 @@
...
+
+WORKDIR /opt/python-gitlab
+COPY . .
+RUN python setup.py bdist_wheel
+
+FROM python:alpine
+
+WORKDIR /opt/python-gitlab
+COPY --from=build /opt/python-gitlab/dist dist/
+RUN pip install $(find dist -name *.whl)
```
Comment (zoidbergwill) [inline Dockerfile]: We could also delete the wheel here, and build the docker image with [`--squash`](https://docs.docker.com/engine/reference/commandline/build/#squash-an-images-layers---squash-experimental), to save a tiny bit of extra space.

File: Dockerfile [hunk 2]
```diff
@@ -0,0 +1,15 @@
+FROM python:alpine AS build
```
Comment (zoidbergwill) [inline Dockerfile]: Should we pin to a Python version?

File: README.rst [hunk 3]
```diff
@@ -41,6 +41,27 @@ Install with pip
...
+
+How to build
+------------
+
+``docker build -t python-gitlab:TAG .``
+
+How to use
+----------
+
+``docker run -it --rm  -e GITLAB_PRIVATE_TOKEN=<your token> =/python-gitlab.cfg python-gitlab <command> ...``
```
Comment (zoidbergwill) [inline README.rst]: Ghost ` =/python-gitlab.cfg` ?

File: Dockerfile (hunk 1 above)
Comment (max-wittig) [reply Dockerfile]: `--squash` still seems to be experimental, but otherwise good idea!

File: README.rst (hunk 3 above)
Comment (max-wittig) [reply README.rst]: I copied that from contrib/docker/README.rst and just glanced over it. Nice find.

File: Dockerfile (hunk 2 above)
Comment (max-wittig) [reply Dockerfile]: Might be better. Maybe 3.8 will break something. I don't think so, but best practice.

**Comment by max-wittig** [comment]
@gpocentek Feel free to merge, if you approve :smiley:

**Pull request merged**